lld/COFF/PDB.cpp

```diff
@@ -1735,6 +1735,8 @@ static uint32_t getSecrelReloc(llvm::COFF::MachineTypes machine) {
   case ARMNT:
     return COFF::IMAGE_REL_ARM_SECREL;
   case ARM64:
+  case ARM64EC:
+  case ARM64X:
```

Comment:
Does it really make sense to handle "ARM64X" here?  Even if you're linking for "arm64x", a given section should be either Arm64 (and needs arm64 relocs) or x64 (and needs x64 relocs).

Reply:
An object file can be ARM64X too, see https://reviews.llvm.org/D149087 for the context of allowing them:

> We do expect ARM64X on input as well. Individual input object file with its machine value of ARM64X are rare and weird, but they do exist and llvm-cvtres can produce them with https://reviews.llvm.org/D148517 (...) According to my testing, ARM64X objects are accepted by link.exe for all targets: ARM64, ARM64EC and ARM64X.

Output of `llvm-cvtres` is not going to cause references errors, so it's indeed not very interesting in this case (it would require somehow generating such file), but adding it here for completeness seems simple enough to me.

Reply:
Thinking some more about it, this is a common pattern. We could reduce the places that need to explicitly care about about ARM64EC/ARM64X with a helper like `getMachineArchType()` and using `Triple::ArchType` type for the switch. I will prepare a separate PR to discuss it.

Reply:
I created #87370 with mentioned clean up. On top of that, this PR could look like this: https://github.com/cjacek/llvm-project/commit/e8e0b1cd41bdcfb39e8e42f26da0ec43b2839606